### PR TITLE
Buttons event

### PR DESCRIPTION
**Pull request by deplanty**
# Modify the button behavior
- Use the pygame event request
- Update the buttons use in the viewers and controllers

### CODE REVIEW

File: src/button/button.py [hunk 1]
```diff
@@ -1,188 +1,219 @@
+# ============================================================================
+# =
+# = Imports
+# =
+# ============================================================================
```
Comment (ldehaine) [inline src/button/button.py]: No. Imports are at the top of the file, we know that.

File: src/controller/game_controller.py [hunk 2]
```diff
@@ -220,3 +229,49 @@ def play_one_round(self):
...
+        """
+
+        print("btn_double")
+
+    def btn_quit(self):
+        """
+        Manage quit on quit button
+        """
+
+        print("btn_quit")
```
Comment (ldehaine) [inline src/controller/game_controller.py]: PEP8: Missing empty line at end of file

File: src/views/view_menu.py [hunk 3]
```diff
@@ -57,29 +62,50 @@ def move_btn_index(move: Moves, btn_index: int, btns_list: List[Button]):
...
+        b = Button(
+            window=window,
+            text=text,
+            pos=(menu_buttons[iid]["x"], menu_buttons[iid]["y"]),
+            size=(200, 50)
+        )
+        b.signal.attach(cmd)
+        btns_list.append(b)
+
+    # btns_list = [btn_play, btn_options, btn_quit]
```
Comment (ldehaine) [inline src/views/view_menu.py]: Do we need this comment?

File: src/button/button.py [hunk 4]
```diff
@@ -1,188 +1,219 @@
...
+    def handle_event(self, event):
         """
-        Return True is the pos is in the Button
+        Handle the events used by the button
+
+        :param pygame.event event: event generated by pygame
         """
 
-        return self.rect.collidepoint(pos)
+        # If button is clicked
```
Comment (ldehaine) [inline src/button/button.py]:
```python
if event.type == pygame.MOUSEMOTION:
```
Here the button isn't clicked, so the comment is false

File: src/button/button.py [hunk 5]
```diff
@@ -1,188 +1,219 @@
...
 
-        if self.background is not None:
-            pygame.draw.rect(window, self.background, self.rect)
+        self.bg = {
+            State.normal: self.bg_normal,
+            State.hover: self.bg_hover,
+            State.pressed: self.bg_pressed
+        }.get(self.state, None)
+
+        if self.bg is not None:
```
Comment (Hboni) [inline src/button/button.py]: Why not just `if self.bg` ??

**Review by ldehaine — CHANGES_REQUESTED** [review]

**Review by ldehaine — APPROVED** [review]

**Pull request merged**